test/matlab_experimental/ctTestThermo.m

```diff
@@ -0,0 +1,482 @@
+                    {0, 'O', 'H'}, {0, 'H', 'AR'}, {0, 'Ar', 'HO2'}};
+            for i = 1:length(data)
+                n = data{i}{1};
+                element = data{i}{2};
+                species = data{i}{3};
+                mElem = self.phase.elementIndex(element);
+                kSpec = self.phase.speciesIndex(species);
+                n1 = self.phase.nAtoms(species, element);
+                n2 = self.phase.nAtoms(kSpec, mElem);
+
+                self.assumeEqual(n1, n);
+                self.assumeEqual(n2, n);
+
+                self.getInvalidValue('nAtoms', {'C', 'H2'}, 'no such species');
+                self.getInvalidValue('nAtoms', {'H', 'CH4'}, 'no such element');
+            end
+        end
+
+        function testElementalMassFraction(self)
+            % Fails because error messages for elementalMassFraction are incorrect.
+            self.assumeFail();
+
+            self.phase.Y = 'H2O:0.5, O2:0.5';
+            Zo = self.phase.elementalMassFraction('O');
+            Zh = self.phase.elementalMassFraction('H');
+            Zar = self.phase.elementalMassFraction('Ar');
+
+            exp1 = 0.5 + 0.5 * (15.999 / 18.015);
+            exp2 = 0.5 * (2.016 / 18.015);
+            exp3 = 0.0;
+
+            self.assumeEqual(Zo, exp1, 'AbsTol', self.atol);
+            self.assumeEqual(Zh, exp2, 'AbsTol', self.atol);
+            self.assumeEqual(Zar, exp3, 'AbsTol', self.atol);
+
+            self.getInvalidValue('elementalMassFraction', {'C'}, 'No such element');
+            self.getInvalidValue('elementalMassFraction', {5}, 'No such element');
+        end
+
+        function testWeights(self)
+            aw = self.phase.atomicMasses;
+            mw = self.phase.molecularWeights;
+
+            self.assumeEqual(length(aw), self.phase.nElements);
+            self.assumeEqual(length(mw), self.phase.nSpecies);
+
+            for i = 1:length(mw)
+                testWeight = 0.0;
+                for j = 1:length(aw)
+                    testWeight = testWeight + ...
+                                 aw(j) * self.phase.nAtoms(i, j);
+                end
+                self.assumeEqual(testWeight, mw(i), 'RelTol', self.rtol);
+            end
+        end
+
+        function testCharges(self)
+            chargePhase = Solution('gri30_ion.yaml', 'gas');
+            charges = chargePhase.charges;
+            test = {{'E',-1}, {'N2',0}, {'H3O+',1}};
+
+            for i = 1:length(test)
+                species = test{i}{1};
+                charge = test{i}{2};
+
+                flag = sum(ismember(chargePhase.speciesNames, species));
+                self.assumeGreaterThan(flag, 0);
+                
+                idx = chargePhase.speciesIndex(species);
+                self.assumeEqual(charges(idx), charge);
+            end
+            clear chargePhase
+        end
+
+        function testReport(self)
+            % Fails because report method is missing. 
+            self.assumeFail();
```

Comment:
Rather than providing the reason for the failure as a comment, it appears that you can also pass it as the argument to `assumeFail`, which will make it appear in the test log. I would recommend this, since it is analogous to what we do in the other test suites.
```suggestion
            self.assumeFail('Fails because report method is missing.');
```